src/App.tsx

```diff
@@ -15,33 +16,100 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [goods, setGoods] = useState<string[]>([]);
+  const [isStart, setStart] = useState(false);
+  const lengthGoods = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setGoods([...goodsFromServer]
+      .filter(good => good.length >= +event.currentTarget.value));
+  };
+
+  const resetGoods = () => {
+    const defaulSelect = document.getElementById('defaultValue');
+
+    setGoods(goodsFromServer);
+    if (defaulSelect) {
+      defaulSelect.setAttribute('selected', '');
+      defaulSelect.removeAttribute('selected');
+    }
```

Comment:
what is this for?

Reply:
For reset <select>.
I add the select attribute to the first option to set it by default, after which I fix it so that the next time when I press reset, the option will be set to default "1" again